app/src/main/java/org/greenstand/android/TreeTracker/models/Locations.kt

```diff
@@ -148,8 +148,9 @@ class LocationDataCapturer(
     private var locationsDeque: Deque<Location> = LinkedList<Location>()
     var generatedTreeUuid: UUID? = null
         private set
-    var convergence: Convergence? = null
+    var lastConvergenceWithinRange: Convergence? = null
```

Comment:
By caching the last convergence that confirmed to the threshold values, we can be sure we will have at least one valid longitude/latitude mean values. Otherwise we have the possibility that GPS data wandered and now is out of the threshold variance range.